content/js1/sprints/3/prep/index.md

```diff
@@ -8,4 +8,634 @@ backlog= 'Module-JS1'
+
+So provided we have an expression for the hours from the time, we can write a conditional statement as follows:
+
+```js
+if (expressionsForHours < 12) {
+  // do code to format the 12 hours
+}
+```
+
+To complete the logic, we now have to achieve a sub-goal:
+
+> 🎯 Sub-goal: Find an expression for the hours from the `time` input
+
+### 🧶 Accessing strings
+
+Given a `time` string we need to access the _first 2 characters_ of the string which represent the hours.
+
+Strings are **zero-indexed**.
+**Index** means **position**, so **zero-indexed** means we start counting character positions from `0` onwards.
+
+Here are the positions/indexes for `"23:00"`
+
+|   index   |   0   |   1   | 2     | 3     | 4     |
+| :-------: | :---: | :---: | ----- | ----- | ----- |
+| character | `"2"` | `"3"` | `":"` | `"0"` | `"0"` |
+
+In JavaScript, we can use square bracket notation to access specific characters in the string using the index.
+
+```js
+time[0]; // evaluates to "2"
+time[1]; // evaluates to "3"
+time[2]; // evaluates to ":"
+// etc
+```
+
+However at the moment, square bracket access wil only give a single character.
+So we must use another method to extract _multiple_ characters from the given string.
+
+### ✂️ Extracting a slice
+
+We can use a function called `slice` to extract 1 or more characters from a string.
+
+[`slice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) is a function that can take 2 arguments: a start index and an end index. `slice` will return a section of the string from the start index up to but not including the start index.
+
+```js
+time; // holds the value "23:00"
+```
+
+```js
+time.slice(0, 2); // will access the characters below
+```
+
+|   index   |   0   |   1   |
+| :-------: | :---: | :---: |
+| character | `"2"` | `"3"` |
+
+So `time.slice(0,2)` will evaluate to `"23"` when the `time` is `"23:00"`. Finally we must convert `"23"` to the number `23`, otherwise we can't compare this value properly. We can use the `+` operator to convert strings to numbers by writing the following:
+
+```js
++time.slice(0, 2); // evaluates to 23
+```
+
+### 🏗️ Assembling the parts
+
+Earlier we had an `if` statement in the form:
+
+```js
+if (expressionForHours > 12) {
+}
+```
+
+So now we've found an expression for the `hours`, we can write:
+
+```js
+if (+time.slice(0, 2) > 12) {
+}
+```
+
+If the time is `"23:00"` then the condition `+time.slice(0,2) > 12` will evaluate to `true` and the code inside the curly braces will be executed.
+
+This `if` statement is implementing the following part of the diagram from earlier:
+
+```mermaid
+
+flowchart TD
+
+A{Check: Are the hours greater than 12?}
+
+```
+
+Now we can format the string using our approach from earlier:
+we’ll need to append `"pm"` to the string expression and subtract 12 from the hours. So we get the following:
+
+```js
+if (+time.slice(0, 2) > 12) {
+  return `${time.slice(0, 2) - 12}:00 pm`;
```

Comment:
I don't think until now we've introduced the idea that a `return` statement _also stops function execution_. So I don't think it's obvious _what to do with this new `if` statement.

I think we either need to introduce here the idea that you can early-`return`, or introduce the idea of an `else` statement.

I think the leap, particularly if this is "do on your own at home" prep work, of what to do with this `if` statement in relation to the existing code, is too big a leap, without one of those two concepts.